FWCore/Common/interface/ProcessBlockHelper.h

```diff
@@ -0,0 +1,120 @@
+  public:
+    ProcessBlockHelperBase const* topProcessBlockHelper() const override;
+    std::vector<std::string> const& topProcessesWithProcessBlockProducts() const override;
+    unsigned int nProcessesInFirstFile() const override;
+    std::vector<std::vector<unsigned int>> const& processBlockCacheIndices() const override;
+    std::vector<std::vector<unsigned int>> const& nEntries() const override;
+    std::vector<unsigned int> const& cacheIndexVectorsPerFile() const override;
+    std::vector<unsigned int> const& cacheEntriesPerFile() const override;
+    unsigned int processBlockIndex(std::string const& processName, EventToProcessBlockIndexes const&) const override;
+    unsigned int outerOffset() const override;
+
+    bool initializedFromInput() const { return initializedFromInput_; }
+
+    bool firstFileDropProcessesAndReorderStored(std::vector<std::string>& storedProcesses,
+                                                std::vector<unsigned int>& storedCacheIndices,
+                                                std::set<std::string> const& processesToKeep,
+                                                std::vector<unsigned int> const& nEntries,
+                                                std::vector<unsigned int>& finalIndexToStoredIndex) const;
+
+    bool dropProcessesAndReorderStored(std::vector<std::string>& storedProcesses,
+                                       std::vector<unsigned int>& storedCacheIndices,
+                                       std::set<std::string> const& processesToKeep,
+                                       std::vector<unsigned int> const& nEntries,
+                                       std::vector<unsigned int>& finalIndexToStoredIndex,
+                                       std::vector<std::string> const& firstFileFinalProcesses) const;
+
+    void initializeFromPrimaryInput(StoredProcessBlockHelper const& storedProcessBlockHelper);
+
+    void fillFromPrimaryInput(StoredProcessBlockHelper const& storedProcessBlockHelper,
+                              std::vector<unsigned int>&& nEntries);
+
+    void clearAfterOutputFilesClose();
+
+  private:
+    void dropProcessesAndReorderStoredImpl(std::vector<std::string>& storedProcesses,
+                                           std::vector<unsigned int>& storedCacheIndices,
+                                           std::vector<std::string>& finalProcesses,
+                                           std::vector<unsigned int> const& nEntries,
+                                           std::vector<unsigned int> const& finalIndexToStoredIndex) const;
+
+    void fillFromPrimaryInputWhenNotEmpty(std::vector<std::string> const& storedProcesses,
+                                          std::vector<unsigned int> const& storedCacheIndices,
+                                          std::vector<unsigned int>&& nEntries);
+
+    void fillEntriesFromPrimaryInput(std::vector<unsigned int>&& nEntries);
```

Comment:
Why use of `&&`? Value semantics is more in line with present C++ best practices.

Reply:
Done. I fixed three functions in this file with && arguments. One is now by value and the other two I changed to const&.